### DISCUSSION TITLE
How i can draw line on chart, can you show simple code ?

### DISCUSSION THREAD

**Discussion by koshak01** [klinecharts/KLineChart#515] [Q&A] [answered]
```

var figure = klinecharts.getFigureClass('line')
        var attrs = {
            coordinates : [
                 {x : 0, y: 0}
                ,{x : 100, y: 100}
            ]
        }
        var styles = {
             style : "dashed"
            ,color : "#f99658"
            ,size : 4
            ,dashedValue: [6,6]
        }
        var ctx = window.klines.getContext('2d') // !!!!! where i can get ctx? 
        new figure({ attrs, styles}).draw(ctx)
```

**Comment by char101** [comment]
Elements in the chart are called overlay and indicator. Overlays are drawn based on coordinates while indicators are drawn based on the data values. Overlays and indicators can use prebuilt figures to draw the shapes or you can draw directly in the canvas using the `draw` method when registering custom overlay/indicators. `ctx` is a parameter in the `draw` method.

To draw a line you create a segment overlay. This can be done interactively or programatically (by giving the `points` value in the `createOverlay` options).

```javascript
chart.createOverlay({
  name: 'segment',
  points: [
    {timestamp: ..., value: ...},
    {timestamp: ..., value: ...},
  ]
}, 'candle_pane')
```

**Comment by koshak01** [comment]
Super, success, can i place not segment, i want put price line

<img width="1454" alt="image" src="https://github.com/klinecharts/KLineChart/assets/3466259/e05a7307-e6c5-419f-95a3-acfb13cd2292">


thx

**Comment by koshak01** [comment]
<img width="873" alt="image" src="https://github.com/klinecharts/KLineChart/assets/3466259/15056f7a-25e2-42bf-9c9c-7be964c19a8e">


Thank you so much, there are simply no words! everything works, I'm a cretin, I don't understand how it works!

**Comment by koshak01** [comment]
<img width="867" alt="image" src="https://github.com/klinecharts/KLineChart/assets/3466259/8ae3ad11-c77a-4e00-b6c0-ae242ffe6d6e">



look 0.058366176 - price with label( create as in example - https://klinecharts.com/en-US/sample/overlay.html)

0.05862904 - create line as you say in this issuse. 

how i can draw label on scale pane( 0.05862904 ) ? 

thx

**Comment by char101** [comment]
Register custom [price line](https://github.com/klinecharts/KLineChart/blob/main/src/extension/overlay/priceLine.ts) overlay, add another text figure to the overlay, set the `x` coordinate of the text to `bounding.width`.

See how the last price line is drawn [here](https://github.com/klinecharts/KLineChart/blob/main/src/view/CandleLastPriceLabelView.ts).

**Comment by koshak01** [comment]
https://github.com/klinecharts/KLineChart/blob/54144d7894f793264125af28603f455e71a759a7/src/extension/overlay/priceLine.ts#L19

i found, tnx

can i be you sponsor?

**Comment by koshak01** [comment]
set the x coordinate of the text to bounding.width - i not see marker ) 

<img width="856" alt="image" src="https://github.com/klinecharts/KLineChart/assets/3466259/3d4a3ddb-1700-4110-8e61-51e384fa2652">


```
klinecharts.registerOverlay({
             name: 'price_line'
            ,totalStep: 2
            ,zLevel : 10000
            ,needDefaultPointFigure: true
            ,needDefaultXAxisFigure: true
            ,needDefaultYAxisFigure: true
          ,createPointFigures: ({ coordinates, bounding, precision, overlay, thousandsSeparator }) => {
            const { value = 0 } = (overlay.points)[0]
            return [
              {
                type: 'line',
                attrs: { coordinates: [coordinates[0], { x: bounding.width, y: coordinates[0].y }] }
              },
              {
                type: 'text',
                ignoreEvent: true,
                attrs: {
                    x: bounding.width,
                    y: coordinates[0].y,
                    text: klinecharts.utils.formatThousands(value.toFixed(precision.price), thousandsSeparator),
                    baseline: 'middle',
                 //   textAlgin: 'right'
                }
                ,styles:{
                    backgroundColor : (overlay.styles).line.color
                  }
              }
            ]
          }
        })

```

**Comment by koshak01** [comment]
<img width="1163" alt="image" src="https://github.com/klinecharts/KLineChart/assets/3466259/68dc3f49-77c1-488a-84a8-5bdd6206c696">


cool ) 

```
,createPointFigures: ({ coordinates, bounding, precision, overlay, thousandsSeparator }) => {
            const { value = 0 } = (overlay.points)[0]
            return [
              {
                    type: 'line'
                  ,attrs: { coordinates: [coordinates[0], { x: bounding.width, y: coordinates[0].y }] }
              }
              ,{
                  type: 'text'
                , ignoreEvent: true
                , visible : true
                , attrs: {
                    x: bounding.width - 100,
                    y: coordinates[0].y,
                    text: klinecharts.utils.formatThousands(value.toFixed(precision.price), thousandsSeparator),
                    baseline: 'middle',
                }
                ,styles:{
                         borderStyle : (overlay.styles).line.style
                        ,borderColor : (overlay.styles).line.color
                        ,borderSize : (overlay.styles).line.size
                        ,style : 'stroke_fill'
                        ,color : 'black'
                        ,backgroundColor: 'white'
                  }
              }
            ]
          }

```

**Comment by koshak01** [comment]
<img width="207" alt="image" src="https://github.com/klinecharts/KLineChart/assets/3466259/25117409-908b-46b3-8693-1c4e0475a472">


i try any types, price label not view, she under YaXis panel ( ((( 


pleasy help

**Reply by ct152098** [reply]
May I ask if it has been solved?

**Comment by koshak01** [comment, marked as answer]
I Register custom [price line](https://github.com/klinecharts/KLineChart/blob/main/src/extension/overlay/simpleTag.ts) overlay!

ales good
